### ISSUE TITLE
Compatibility with Bootstrap Icons

### ISSUE THREAD

**Issue by lamxw2** [maxmarcon/live_select#56]
Hi there, thank you for creating this library!

I'm in a little tricky situation where I can't make use of `tailwind` and `daisyui`, so I have to handle styling based on the css my team uses, mainly bootstrap.

Without the usage of `tailwind` or `daisyui`, the `X` icon on tags appears as a rectangle, like this: 
![image](https://github.com/maxmarcon/live_select/assets/13754413/dd5f9c2d-968f-4e28-b764-2ca56c36b3a7)
(it looks the same even when style is set to `:none`)

I have set `bi bi-x` (https://icons.getbootstrap.com/icons/x/) to `clear_button_class`, along with another custom class for `background: transparent` and `border-style: none`, which results in this:
![image](https://github.com/maxmarcon/live_select/assets/13754413/90d23f41-732b-4e6d-a169-221cf973fd5d)

This is what appears on inspect element of the x's
![image](https://github.com/maxmarcon/live_select/assets/13754413/b11fbb6c-d0f0-4d95-949c-9a4ea028d8a1)

The `svg` is present regardless of the bootstrap class and default styling (as it still appears with `style={:none}`), it is an X when `bi bi-x` is used, and as a rectangle when nothing is used. 

Google tells me that duplicates tend to happen when either bootstrap-icons is installed with another icon library, or when `<i class="bi bi-x"/>` is used instead of `<i class="bi bi-x"></i>`. I checked our packages and we don't appear to have another icon library installed.

Workarounds that I have tried and failed:
- Setting `content: ''` on `bi::before`
  - The bootstrap x appears in `bi::before`, so it just resulted in the x disappearing entirely in both
- Adding the bootstrap icon to the `:tag` slot and hiding the clear button
  - This didn't work because the hidden clear button leaves a transparent right "padding". This also doesn't work as the X icon itself isn't clickable, since it's not a button.

I'm rather new to LiveView (I'm also primarily a backend dev, not frontend), so this could also just be that I didn't import/use the bootstrap icon properly. Otherwise, it may be due to an incompatibility of using it directly in a `button` class?

Hopefully this makes sense!

Edit: I've found a workaround by making use of `display: grid`, having the tag contents all be in 1 row, then setting `display: flex` on the tag contents

**Comment by maxmarcon** [comment]
I added a new slot `:clear_button` that you can use to define your own clear button. If left unspecified, it defaults to the current "x" icon.

So in your case, if you want to remove the content of the button completely, this should work:

```
<.live_select>
  <:clear_button></:clear_button>
</.live_select>
```

It's on main, please point your live_select dependency to main (`{:live_select, github: "maxmarcon/live_select"}`) and give it a try. Thanks

**Comment by lamxw2** [comment]
Thank you, I'll give that a try!

**Comment by lamxw2** [comment]
Hi @maxmarcon , unfortunately our app is on Elixir 1.13 and we are using Phoenix 1.6, so we cannot upgrade to `phoenix_html` 4.0, which the latest version of LiveSelect uses 🥲 It is unfortunately not up to me to upgrade our elixir and Phoenix versions. Thank you for your time though, it is much appreciated!

**Comment by maxmarcon** [comment]
Closed by 23d3976
